### PR TITLE
Allow any number of digits in the secfrac part of an ISO8601 date

### PR DESCRIPTION
**Pull request by cysp**
@TabDigital/ping-api

### CODE REVIEW

File: lib/matchers/isoDate.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 var _ = require('lodash');
 var factory = require('../factory');
 
-var dateTimeRegex   = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d\d\d)?(Z|[+-]\d\d:\d\d)?$/;
+var dateTimeRegex   = /^\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d(.\d+)?(Z|[+-]\d\d:\d\d)?$/;
```
Comment (anklos) [inline lib/matchers/isoDate.js]: this can be written a bit more clean as `^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(.\d{3,4})?(Z|[+-]\d{2}:\d{2})?$`

Comment (srushti) [inline lib/matchers/isoDate.js]: Except the part where you've put `(.\d{3,4})` for the milliseconds, which should remain `(.\d+)` as @cysp put it.

Comment (cysp) [inline lib/matchers/isoDate.js]: That `.` should be escaped too, didn't notice that before.

**Comment by rprieto** [comment]
Looks good to me! :+1: Can you just squash all commits?

**Comment by cysp** [comment]
I doubt I'm going to come back to this, and also it looks like it was fixed upstream _two years ago_. 😬

**Pull request closed without merging**